File: tests/test-physic/src/main/java/pize/tests/physic/DynamicRect.java
package pize.tests.physic;

import pize.physic.BoundingBox2;
import pize.physic.RectBody;
import pize.physic.Velocity2f;

public class DynamicRect extends RectBody{

    private final Velocity2f velocity;

    public DynamicRect(BoundingBox2 rect){
        super(rect);

        velocity = new Velocity2f();
    }

    public Velocity2f motion(){
        return velocity;
    }

}

File: tests/test-physic/src/main/kotlin/pize/tests/physic/DynamicRect.kt
package pize.tests.physic

import pize.physic.BoundingBox2
import pize.physic.RectBody
import pize.physic.Velocity2f

class DynamicRect(rect: BoundingBox2?) : RectBody(rect) {
    private val velocity: Velocity2f

    init {
        velocity = Velocity2f()
    }

    fun motion(): Velocity2f {
        return velocity
    }
}
